### DISCUSSION TITLE
Basic 1D case

### DISCUSSION THREAD

**Discussion by MaAl13** [mmuckley/torchkbnufft#88] [Q&A]
Hello, i have several 1D time series data that is non-uniformly sampled. How would i apply this case in your package? Thank you very much for the help in advance :)

**Comment by mmuckley** [comment]
Converting to a discussion as this does not document a bug.

Here's a way you could do it for a 64-length signal with a batch size of 5:

```python
import torchkbnufft as tkbn
import torch

signal = torch.randn(5, 1, 64) + 1j * torch.randn(5, 1, 64)
omega = (torch.rand(1, 55) - 0.5) * 2 * torch.pi # in radians/sample

nufft_ob = tkbn.KbNufft(im_size=(64,))

frequency_estimates = nufft_ob(signal, omega)
```

**Reply by mmuckley** [reply]
@MaAl13

**Reply by awjang** [reply]
Hi, thank you for this!

I too am trying to implement a 1D case where I have a train of length 64 k-space data, each of which requires regridding.

Since I measure my gradients, I can calculate the corresponding k-trajectory, so my approach will be to first density compensating acquired data (calc_density_compensation_function) followed by regridding (KbInterpAdjoint) as I am working in k-domain.

Can you please confirm the following?
- In the above, omega should be (torch.rand(1, 64) - ..., correct?
- To convert k-trajectory to radians per sample, I should do 2*pi*gamma*cumtrapz(G)*dt*dx, where dx is pixel size in [cm], G is gradient in [G/cm] and gamma is [Hz/G/cm]?
- For the 1D case, it seems that you don't have to define grid_size (which is typically 1.25 to 2 times im_size)? I noticed if grid_size is not set, the interpolated output is 2x the length of im_size?

**Reply by mmuckley** [reply]
Hello @awjang.

- `omega` should be in radians and include the radians conversion.
- This is more of an MRI fundamentals question than something for the package, and I don't remember the exact formulas without a reference. However, from what you've written I think you're calculating the trajectory in radians/cm, whereas the package will require that you convert that to radians/voxel.
- Yes that's correct. The main reason is that due to approximation errors with how the NUFFT is done, it's much better to do it on an oversampled grid to maintain accuracy.

**Reply by awjang** [reply]
Thanks for the quick reply. Regarding bullet 3, the oversampling is more dense sampling as opposed to sampling farther out in k, yes? Basically what's typically done when acquiring along the readout in the MRI scanner.
